core/amber/src/main/scala/edu/uci/ics/texera/web/resource/ResultDownloadResource.scala

```diff
@@ -148,7 +138,10 @@ object ResultDownloadResource {
 
     // upload the content asynchronously to avoid long waiting on the user side.
     // may change to thread pool
-    new Thread(() => uploadResult(sheetService, sheetId, result)).start()
+    new Thread(() => {
```

Comment:
better to change to a thread pool solution now rather than later. It's very simple, just use an `Executors.newFixedThreadPool`

Reply:
done